code/datums/keybindings/carbon.dm

```diff
@@ -22,6 +22,15 @@
 	var/mob/living/carbon/M = C.mob
 	M.throw_mode_off()
 
+/datum/keybinding/carbon/give_item
+	name = "Give Item (Toggle)"
```

Comment:
the semantics of the (toggle) is weird, because contrary to what i expected originally, if you:
- click the toggle (turning it on)
- click the person (giving them the prompt)
- click the toggle again 

then you won't STOP giving them the thing - you won't "toggle giving off".

Swapping hands turning gifting off makes sense, kinda, but it's a bit of a confusing UI.
Would it be possible to have the gift UI button have three (rather than two) states of outline - off / target selection /  waiting for the target to accept? And clicking on it cycling the mode as
- off -> target selection
- target selection -> off
- waiting -> off

(but i also expect that this is a bit too much works, so eh)

Reply:
I'm hoping players realize that after offering an item, having their normal mouse return along with the chat feedback means they're no longer in a giving state. Adding the third HUD icon state seems a bit odd.

Reply:
> players realize that after offering an item, having their normal mouse return along with the chat feedback means they're no longer in a giving state.

They will realize, and that's the problem. Because they actually *are* still in a giving state until either the offer is accepted or the user swaps hands. But there's no visual distinction between this "giving state" and regular "just walking around doing ss13 things" one.
Also note how we you're using term "giving state" to mean "has click_intercept", whereas I use it to mean [effectively] `HAS_TRAIT(user, TRAIT_OFFERING_ITEM)`

Okay, constructively, UI-wise: How about the give button has only two states (on and off), but it becomes "off" not when the click_intercept is destroyed, but when the user loses TRAIT_OFFERING_ITEM / when `
/obj/screen/alert/take_item` is destroyed? (Or on manual click, which would do the same thing of canceling offer)

Reply:
I see what you're saying now. I think having the give icon also double as notification as to whether or not you're in a giving state might be confusing, I dunno. I'm thinking maybe I could make an alert for the giver which shows two people exchanging something as a representation that they're in a giving state.